src/renderers/WebGLRenderer.js

```diff
@@ -1053,6 +1053,37 @@ function WebGLRenderer( parameters = {} ) {
 
 	};
 
+	this.renderShadowMap = function ( lightsWithShadows, scene, camera ) {
+
+		currentRenderState = renderStates.get( scene, renderStateStack.length );
+		currentRenderState.init();
+
+		renderStateStack.push( currentRenderState );
+
+		// We don't need a full init of currentRenderState - we're just updating the lights specified in lightsWithShadows.
+
+		_localClippingEnabled = this.localClippingEnabled;
+		_clippingEnabled = clipping.init( this.clippingPlanes, _localClippingEnabled, camera );
+
+		if ( _clippingEnabled === true ) clipping.beginShadows();
+
+		shadowMap.needsUpdate = true;
+		shadowMap.render( lightsWithShadows, scene, camera, clipping );
+
+		if ( _clippingEnabled === true ) clipping.endShadows( camera );
+
+		if ( renderStateStack.length > 0 ) {
```

Comment:
It seems `renderStateStack.pop();` is missing right before this `if` statement.

Reply:
Good catch, thanks!